src/pages/portfolio/utils.ts

```diff
@@ -0,0 +1,63 @@
+import { ProjectType } from '../../views/portfolio-page/types';
+
+const maxNumberOfTabs = 3;
+const otherLabel = 'More';
+const allLabel = 'All';
+
+type GroupedProjectsByLabelType = {
+  projectLabel: string;
+  projectProperties: ProjectType[];
+};
+
+const compareLengthOfProjects = (
+  longerLengthGroupedProjectsByLabel: GroupedProjectsByLabelType,
+  shorterLengthGroupedProjectsByLabel: GroupedProjectsByLabelType,
+): number => {
+  return (
+    shorterLengthGroupedProjectsByLabel.projectProperties.length -
+    longerLengthGroupedProjectsByLabel.projectProperties.length
+  );
+};
+
+const groupBy = (projects: ProjectType[]) => {
+  return projects.reduce((groupedProjects, project) => {
+    // eslint-disable-next-line no-param-reassign
+    (groupedProjects[project.projectLabel] = groupedProjects[project.projectLabel] || []).push(project);
+    return groupedProjects;
+  }, {} as { [key: string]: ProjectType[] });
```

Comment:
Wouldn't it work if you write `reduce<{ [key: string]: ProjectType[] }>(...)` instead of this type casting?